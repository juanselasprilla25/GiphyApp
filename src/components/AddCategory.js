import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setcategory }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInput = (e) => {
    setinputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setcategory((cat) => [inputValue, ...cat]);
      setinputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='CategorÃ­a'
        onChange={handleInput}
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setcategory: PropTypes.func.isRequired,
};

AddCategory.defaultProps = {};
