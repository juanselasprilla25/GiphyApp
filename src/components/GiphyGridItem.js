import React from "react";
import PropTypes from "prop-types";

export const GiphyGridItem = ({ title, url }) => {
  return (
    <div className='card animate__animated animate__bounceInLeft'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GiphyGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
