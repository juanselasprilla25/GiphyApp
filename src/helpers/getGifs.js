export const getGifs = async (category) => {
  try {
    const ApiKey = "2ES5pAo98fC0I8jw5KJI2HJzvMt2q5eQ";
    const url = `https://api.giphy.com/v1/gifs/search?api_key=${ApiKey}&limit=10&q=${encodeURI(
      category
    )}`;

    const response = await fetch(url);

    const { data } = await response.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    return gifs;
  } catch (error) {
    return error;
  }
};
