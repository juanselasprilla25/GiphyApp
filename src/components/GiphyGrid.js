import React from "react";

/**
 * Components
 */
import { GiphyGridItem } from "./GiphyGridItem";

/**
 * Custom Hooks
 */
import { UseFetchGifs } from "../hooks/useFetchGifs";

export const GiphyGrid = ({ category }) => {
  const { data: Images, loading } = UseFetchGifs(category);
  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>
      {loading && (
        <p className='animate__animated animate__flash'>Loading Data...</p>
      )}
      <div className='cardGrid'>
        {/* <ol> */}
        {Images.map((image) => {
          return <GiphyGridItem key={image.id} {...image} />;
        })}
        {/* </ol> */}
      </div>
    </>
  );
};
