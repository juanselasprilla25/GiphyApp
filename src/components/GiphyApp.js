import React, { useState } from "react";

/**
 * Components
 */
import { AddCategory } from "./AddCategory";
import { GiphyGrid } from "./GiphyGrid";

export const GiphyApp = () => {
  const [category, setcategory] = useState(["One Punch"]);
  return (
    <>
      <h2>GiphyApp</h2>
      <hr />
      <AddCategory setcategory={setcategory} />
      {/* <ol> */}
      {category.map((categoryItem) => {
        return <GiphyGrid key={categoryItem} category={categoryItem} />;
      })}
      {/* </ol> */}
    </>
  );
};
