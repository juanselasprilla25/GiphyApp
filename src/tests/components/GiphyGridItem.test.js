import { shallow } from "enzyme";
import { GiphyGridItem } from "../../components/GiphyGridItem";
import { getGifs } from "../../helpers/getGifs";

describe("Testing <GiphyGridItem/> in GiphyGridItem.js file ", () => {
  test("Rendering correctly without props passed in", () => {
    const category = "Trees";
    const imgs = getGifs(category);
    const wrapper = shallow(<GiphyGridItem />);

    expect(wrapper).toMatchSnapshot();
  });
  test("Rendering correctly with 'title' and 'url' props passed in", () => {
    // const testTitle = "Rick";
    const testTitle = 1;
    const testUrl =
      "https://media2.giphy.com/media/SsZViiaRCjgp8fVexU/giphy.gif?cid=79d2ed22nvb0imbtongho7ze8m3bb0xr4vxg596hw32capfq&rid=giphy.gif";
    const wrapper = shallow(<GiphyGridItem title={testTitle} url={testUrl} />);

    expect(wrapper).toMatchSnapshot();
  });
});
