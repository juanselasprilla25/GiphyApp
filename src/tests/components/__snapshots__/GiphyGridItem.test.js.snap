// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GiphyGridItem/> in GiphyGridItem.js file  Rendering correctly with 'title' and 'url' props passed in 1`] = `
<div
  className="card animate__animated animate__bounceInLeft"
>
  <img
    alt={1}
    src="https://media2.giphy.com/media/SsZViiaRCjgp8fVexU/giphy.gif?cid=79d2ed22nvb0imbtongho7ze8m3bb0xr4vxg596hw32capfq&rid=giphy.gif"
  />
  <p>
    1
  </p>
</div>
`;

exports[`Testing <GiphyGridItem/> in GiphyGridItem.js file  Rendering correctly without props passed in 1`] = `
<div
  className="card animate__animated animate__bounceInLeft"
>
  <img />
  <p />
</div>
`;
